src/tasks/managingMiscellaniaActivity.ts

```diff
@@ -0,0 +1,247 @@
+		const { userID, channelID, main_collect, secondary_collect, cofferCost } = data;
+		const user = await mUserFetch(userID);
+		const loot = new Bank();
+
+		const main = main_collect;
+		const secondary = secondary_collect;
+		const cofferMax = 75000;
+
+		const daysDifference = Math.round(cofferCost / 75_000);
+
+		const workerEffectiveness = (cofferMax * 100) / 8333;
+
+		const dailyResourcePoints = Math.floor((workerEffectiveness * 127) / 100);
+
+		const totalResourcePoints = Math.min(262143, daysDifference * dailyResourcePoints);
+
+		const mainQty = getBaseQuantity(main, 10, totalResourcePoints);
+		const secondaryQty = getBaseQuantity(secondary, 5, totalResourcePoints);
+
+		await userStatsUpdate(user.id, { last_managing_miscellania_timestamp: new Date().getTime() }, {});
+
+		function addLoot(category: string, quantity: number) {
+			switch (category) {
+				case 'Wood':
+					loot.add('Maple logs', quantity);
+					loot.add(nestTable.roll(Math.min(999, Math.floor(quantity / 100))));
+					break;
+				case 'Mining':
+					loot.add('Coal', quantity);
+					loot.add(miningGems.roll(Math.floor(quantity / 200 + 0.5)));
+					break;
+				case 'Fishing':
+					loot.add('Raw tuna', Math.floor(0.5 * quantity));
+					loot.add('Raw swordfish', Math.floor(0.15 * quantity));
+					loot.add(fishingLoot.roll(Math.floor(quantity / 200)));
+					break;
+				case 'Cooked Fish':
+					loot.add('Tuna', Math.floor(0.5 * quantity));
+					loot.add('Swordfish', Math.floor(0.15 * quantity));
+					loot.add(fishingLoot.roll(Math.floor(quantity / 200)));
+					break;
+				case 'Herbs':
+					loot.add(herbsLoot.roll(quantity));
+					loot.add(herbSeedsLoot.roll(Math.floor(quantity / 100)));
+					break;
+				case 'Flax':
+					loot.add('Flax', quantity);
+					loot.add(flaxSeedsLoot.roll(Math.floor(quantity / 600)));
+					break;
+				case 'Mahogany':
+					loot.add('Mahogany logs', quantity);
+					loot.add(nestTable.roll(Math.floor(quantity / 350)));
+					break;
+				case 'Teak':
+					loot.add('Teak logs', quantity);
+					loot.add(nestTable.roll(Math.floor(quantity / 350)));
+					break;
+				case 'Hardwood (Mahogany and Teak)':
+					loot.add('Mahogany logs', Math.floor(quantity * 0.5));
+					loot.add('Teak logs', Math.floor(quantity * 0.5));
+					loot.add(nestTable.roll(Math.floor(quantity / 350)));
+					break;
+				case 'Farm':
+					loot.add(seedsLoot.roll(quantity));
+					loot.add(treeSeedsLoot.roll(Math.floor(quantity / 200)));
+					break;
+			}
+		}
+
+		addLoot(main, mainQty);
+		addLoot(secondary, secondaryQty);
+
+		const str = `${user}, ${user.minionName} finished collecting from the kingdom, you received ${loot}.`;
+
+		await user.addItemsToBank({ items: loot, collectionLog: true });
+		await updateBankSetting('managing_miscellania_loot', loot);
+
+		const image = await makeBankImage({ bank: loot, title: 'Managing Miscellania Loot', user });
+
+		handleTripFinish(user, channelID, str, image.file.attachment, data, loot);
+	}
```

Comment:
This should be abstracted outside this task, so that we can test/run it without a minion trip. It might be something like `function determineManagingMiscLoot(inputsNeeded)`. This gives several big benefits: we can add simulating commands for it, and add tests for it.

You may want a second function which determines points/etc based on dates (not 100% sure how it works)